service/metrics/chain.go

```diff
@@ -0,0 +1,64 @@
+// Copyright 2021 Optakt Labs OÜ
+//
+// Licensed under the Apache License, Version 2.0 (the "License"); you may not
+// use this file except in compliance with the License. You may obtain a copy of
+// the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+// WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+// License for the specific language governing permissions and limitations under
+// the License.
+
+package metrics
+
+import (
+	"github.com/onflow/flow-go/model/flow"
+
+	"github.com/optakt/flow-dps/models/dps"
+)
+
+type Chain struct {
+	chain dps.Chain
+	time  Time
+}
+
+func NewChain(chain dps.Chain, time Time) *Chain {
+	c := Chain{
+		chain: chain,
+		time:  time,
+	}
+	return &c
+}
+
+func (c *Chain) Root() (uint64, error) {
+	defer c.time.Duration("root")()
```

Comment:
Neat trick, I like it.

Reply:
Yeah, I was also happy when I thought of it :D